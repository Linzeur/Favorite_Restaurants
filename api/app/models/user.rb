class User < ApplicationRecord

  validates :name, presence: true
  validates :email, format: { with: /[a-zA-Z0-9]*@[a-zA-Z]*.[a-zA-Z0-9]*/, message: "Bad format"  },
                    uniqueness: true

  has_secure_password
  has_secure_token
  has_secure_token :reset_digest

  has_many :locations
  has_many :favorite_restaurants

  def self.valid_login?(email, password)
    user = find_by(email: email)
    user if user&.authenticate(password)
  end

  def invalidate_token
    update(token: nil)
  end
end
