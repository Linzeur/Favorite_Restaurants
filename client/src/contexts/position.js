import React from "react";

const PositionContext = React.createContext();

function PositionProvider(props) {
  const [position, setPosition] = React.useState([0, 0, false]);

  React.useEffect(() => {
    const watchId = navigator.geolocation.watchPosition(({ coords }) => {
      setPosition([coords.latitude, coords.longitude, true]);
      sessionStorage.setItem(
        "lastCurrentPosition",
        JSON.stringify({ lat: coords.latitude, lng: coords.longitude })
      );
    });
    return () => navigator.geolocation.clearWatch(watchId);
  }, [setPosition]);

  const value = {
    latitude: position[0],
    longitude: position[1],
    callAPI: position[2],
    onChange: (lat, lng, callAPI) => {
      setPosition([lat, lng, callAPI]);
    }
  };

  return <PositionContext.Provider value={value} {...props} />;
}

export { PositionProvider, PositionContext };
