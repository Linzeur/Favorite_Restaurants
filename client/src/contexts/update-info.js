import React from "react";

const UpdateListContext = React.createContext();

function UpdateListProvider(props) {
  const [updateList, setUpdateList] = React.useState({
    locations: 0,
    favorites: 0
  });

  const value = {
    locations: updateList.locations,
    favorites: updateList.favorites,
    onChange: key => {
      setUpdateList(prevState => {
        return { ...prevState, [key]: prevState[key] + 1 };
      });
    }
  };

  return <UpdateListContext.Provider value={value} {...props} />;
}

export { UpdateListProvider, UpdateListContext };
