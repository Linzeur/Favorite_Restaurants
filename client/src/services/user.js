import { apiUrl } from "../util";

const API_USERS = `${apiUrl}users`;

async function createUser(userData) {
  const response = await fetch(API_USERS, {
    method: "POST",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(userData)
  });

  if (!response.ok) {
    const { errors } = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
}

export { createUser };
