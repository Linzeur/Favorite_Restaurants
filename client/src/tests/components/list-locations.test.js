import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";

import ListLocations from "../../components/list-locations";
import { UserProvider } from "../../contexts/user";
import { PositionProvider } from "../../contexts/position";
import {
  UpdateListProvider,
  UpdateListContext
} from "../../contexts/update-info";

const user = {
  name: "User Test",
  email: "test@test.com"
};

const listLocations = [
  {
    id: 1,
    name: "Ovalo Miraflores",
    address: "Parque Kennedy",
    latitude: "-12.119109",
    longitude: "-77.029313",
    district: "Miraflores"
  },
  {
    id: 2,
    name: "Cercanias de Larcomar",
    address: "Armendariz",
    latitude: "-12.134923",
    longitude: "-77.025859",
    district: "Miraflores"
  },
  {
    id: 3,
    name: "Casa de mi amigo Paul",
    address: "Calle Jorge Chavez",
    latitude: "-12.121387",
    longitude: "-77.036953",
    district: "Miraflores"
  }
];

test("list-location component without consuming API", () => {
  const setOptionSelectedFn = jest.fn();

  const { asFragment } = render(
    <UpdateListProvider>
      <ListLocations optionSelectedFn={setOptionSelectedFn} />
    </UpdateListProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("list-location component consuming API", async () => {
  localStorage.setItem("user", JSON.stringify(user));
  fetch.mockResponseOnce(JSON.stringify(listLocations));
  const setOptionSelectedFn = jest.fn();

  const { asFragment, getAllByLabelText } = render(
    <UserProvider>
      <UpdateListProvider>
        <UpdateListContext.Consumer>
          {updateInfo => {
            if (updateInfo.locations === 0) {
              updateInfo.onChange("locations");
            }
            return <ListLocations optionSelectedFn={setOptionSelectedFn} />;
          }}
        </UpdateListContext.Consumer>
      </UpdateListProvider>
    </UserProvider>
  );

  let locations;
  await wait(() => {
    locations = getAllByLabelText("Element location");
  });

  expect(asFragment()).toMatchSnapshot();

  expect(locations.length).toBe(3);
});

test("select an option", async () => {
  localStorage.setItem("user", JSON.stringify(user));
  fetch.mockResponseOnce(JSON.stringify(listLocations));
  const setOptionSelectedFn = jest.fn();

  const { getAllByLabelText, asFragment } = render(
    <UserProvider>
      <PositionProvider>
        <UpdateListProvider>
          <UpdateListContext.Consumer>
            {updateInfo => {
              if (updateInfo.locations === 0) {
                updateInfo.onChange("locations");
              }
              return <ListLocations optionSelectedFn={setOptionSelectedFn} />;
            }}
          </UpdateListContext.Consumer>
        </UpdateListProvider>
      </PositionProvider>
    </UserProvider>
  );

  let locations;
  await wait(() => {
    locations = getAllByLabelText("Element location");
  });

  fireEvent.click(locations[0]);

  expect(setOptionSelectedFn).toHaveBeenCalledWith(listLocations[0].id);
});

test("show modal to edit location", async () => {
  localStorage.setItem("user", JSON.stringify(user));
  fetch.mockResponseOnce(JSON.stringify(listLocations));
  const setOptionSelectedFn = jest.fn();

  const { getAllByLabelText, getByText, getByLabelText, asFragment } = render(
    <UserProvider>
      <PositionProvider>
        <UpdateListProvider>
          <UpdateListContext.Consumer>
            {updateInfo => {
              if (updateInfo.locations === 0) {
                updateInfo.onChange("locations");
              }
              return <ListLocations optionSelectedFn={setOptionSelectedFn} />;
            }}
          </UpdateListContext.Consumer>
        </UpdateListProvider>
      </PositionProvider>
    </UserProvider>
  );

  let editButtons;
  await wait(() => {
    editButtons = getAllByLabelText("Edit location");
  });

  fireEvent.click(editButtons[0]);

  expect(asFragment()).toMatchSnapshot();

  const modalEditLocation = getByText("Edit Location");
  const modalInputName = getByLabelText("Enter Name");
  const modalInputAddress = getByLabelText("Enter Address");
  const modalInputDistrict = getByLabelText("Enter District");

  expect(modalEditLocation).not.toBeNull();
  expect(modalInputName.value).toBe(listLocations[0].name);
  expect(modalInputAddress.value).toBe(listLocations[0].address);
  expect(modalInputDistrict.value).toBe(listLocations[0].district);
});
