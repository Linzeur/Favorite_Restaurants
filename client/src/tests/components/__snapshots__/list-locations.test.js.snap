// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-location component consuming API 1`] = `
<DocumentFragment>
  <div
    class="css-1ylqutl-Search"
  >
    <input
      aria-label="Search an option"
      class="css-x0timv-Search"
      id="search"
      name="search"
      placeholder="Search an option"
      type="text"
    />
    <button
      aria-label="Create location"
      class="css-1mgkve5-AddLocation"
      type="button"
    >
      <svg
        class="css-1g31t4l-AddLocation"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </div>
  <ul
    class="css-17f5pel-ListLocations"
  >
    <li
      class="css-70qvj9"
    >
      <div
        aria-label="Element location"
        class="css-j2js4d"
        role="button"
      >
        <svg
          class="css-xbesph"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          />
        </svg>
        <div
          class="css-esxnue"
        >
          <span>
            Ovalo Miraflores
          </span>
          <span
            class="css-2czhfz"
          >
            Miraflores
          </span>
        </div>
      </div>
      <button
        aria-label="Edit location"
        class="css-1ch2zm0"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete location"
        class="css-10jcxoq"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </li>
    <li
      class="css-70qvj9"
    >
      <div
        aria-label="Element location"
        class="css-j2js4d"
        role="button"
      >
        <svg
          class="css-xbesph"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          />
        </svg>
        <div
          class="css-esxnue"
        >
          <span>
            Cercanias de Larcomar
          </span>
          <span
            class="css-2czhfz"
          >
            Miraflores
          </span>
        </div>
      </div>
      <button
        aria-label="Edit location"
        class="css-1ch2zm0"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete location"
        class="css-10jcxoq"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </li>
    <li
      class="css-70qvj9"
    >
      <div
        aria-label="Element location"
        class="css-j2js4d"
        role="button"
      >
        <svg
          class="css-xbesph"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          />
        </svg>
        <div
          class="css-esxnue"
        >
          <span>
            Casa de mi amigo Paul
          </span>
          <span
            class="css-2czhfz"
          >
            Miraflores
          </span>
        </div>
      </div>
      <button
        aria-label="Edit location"
        class="css-1ch2zm0"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete location"
        class="css-10jcxoq"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </li>
  </ul>
  <div
    class="css-gn3t32-Footer"
  >
    <span
      class="css-8ep07x-Footer"
    >
      Results:
    </span>
    <span
      class="css-xp9py7-Footer"
    >
      3 locations
    </span>
  </div>
</DocumentFragment>
`;

exports[`list-location component without consuming API 1`] = `
<DocumentFragment>
  <div
    class="css-1ylqutl-Search"
  >
    <input
      aria-label="Search an option"
      class="css-x0timv-Search"
      id="search"
      name="search"
      placeholder="Search an option"
      type="text"
    />
    <button
      aria-label="Create location"
      class="css-1mgkve5-AddLocation"
      type="button"
    >
      <svg
        class="css-1g31t4l-AddLocation"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </div>
  <div
    class="css-w40flo-Spinner"
  >
    <div
      class="css-lehx5n-Spinner"
    />
    <div
      class="css-wk0iwg-Spinner"
    />
    <div
      class="css-1xsj7ka-Spinner"
    />
  </div>
  <div
    class="css-gn3t32-Footer"
  >
    <span
      class="css-8ep07x-Footer"
    >
      Results:
    </span>
    <span
      class="css-xp9py7-Footer"
    >
      0 locations
    </span>
  </div>
</DocumentFragment>
`;

exports[`show modal to edit location 1`] = `
<DocumentFragment>
  <div
    class="css-1ylqutl-Search"
  >
    <input
      aria-label="Search an option"
      class="css-x0timv-Search"
      id="search"
      name="search"
      placeholder="Search an option"
      type="text"
    />
    <button
      aria-label="Create location"
      class="css-1mgkve5-AddLocation"
      type="button"
    >
      <svg
        class="css-1g31t4l-AddLocation"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </div>
  <ul
    class="css-17f5pel-ListLocations"
  >
    <li
      class="css-70qvj9"
    >
      <div
        aria-label="Element location"
        class="css-j2js4d"
        role="button"
      >
        <svg
          class="css-xbesph"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          />
        </svg>
        <div
          class="css-esxnue"
        >
          <span>
            Ovalo Miraflores
          </span>
          <span
            class="css-2czhfz"
          >
            Miraflores
          </span>
        </div>
      </div>
      <button
        aria-label="Edit location"
        class="css-1ch2zm0"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete location"
        class="css-10jcxoq"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </li>
    <li
      class="css-70qvj9"
    >
      <div
        aria-label="Element location"
        class="css-j2js4d"
        role="button"
      >
        <svg
          class="css-xbesph"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          />
        </svg>
        <div
          class="css-esxnue"
        >
          <span>
            Cercanias de Larcomar
          </span>
          <span
            class="css-2czhfz"
          >
            Miraflores
          </span>
        </div>
      </div>
      <button
        aria-label="Edit location"
        class="css-1ch2zm0"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete location"
        class="css-10jcxoq"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </li>
    <li
      class="css-70qvj9"
    >
      <div
        aria-label="Element location"
        class="css-j2js4d"
        role="button"
      >
        <svg
          class="css-xbesph"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          />
        </svg>
        <div
          class="css-esxnue"
        >
          <span>
            Casa de mi amigo Paul
          </span>
          <span
            class="css-2czhfz"
          >
            Miraflores
          </span>
        </div>
      </div>
      <button
        aria-label="Edit location"
        class="css-1ch2zm0"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete location"
        class="css-10jcxoq"
        type="button"
      >
        <svg
          class="css-3fuzg5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </li>
  </ul>
  <div
    class="css-gn3t32-Footer"
  >
    <span
      class="css-8ep07x-Footer"
    >
      Results:
    </span>
    <span
      class="css-xp9py7-Footer"
    >
      3 locations
    </span>
  </div>
</DocumentFragment>
`;
